### PR TITLE
Authentication capabilities

### PR DESCRIPTION
**Pull request by maxyakimakha**
Update authentication section with authentication capabilities for pre-shared key ease of input and PSK input methods.
Add clarification that each authentication method requires its own specific messages.
Add CDDL message for authentication capabilities.
Remove certificate-fingerprint-pair message and fix description of format used for HKDF info in the proposed default authentication method.

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -401,6 +401,25 @@ Issue(139): Clarify scoping/uniqueness of request IDs.
 Authentication {#authentication}
 ================================
 
+Each supported authentication method is implemeted via authentication messages
+specific to that method.  Authentication method is explicitly specified by
```
Comment (pthatcherg) [inline index.bs]: "Authentication method" => "The authentication method"

File: index.bs [hunk 2]
```diff
@@ -401,6 +401,25 @@ Issue(139): Clarify scoping/uniqueness of request IDs.
...
+the message itself.  Authentication status message is common for all authentication
```
Comment (pthatcherg) [inline index.bs]: "Authentication status message" => "The authentication status message"

File: index.bs [hunk 3]
```diff
@@ -401,6 +401,25 @@ Issue(139): Clarify scoping/uniqueness of request IDs.
...
+methods.  Any new authentication method added must define new authentication messages.
+Default authentication method is a challenge-response authentication with
```
Comment (pthatcherg) [inline index.bs]: "Default authentication method" => "The default authentication method"

File: index.bs [hunk 4]
```diff
@@ -401,6 +401,25 @@ Issue(139): Clarify scoping/uniqueness of request IDs.
...
+auth-request-hkdf-scrypt-psk and auth-response-hkdf-scrypt-psk-result.
+
+Prior to authentication, agents exchange auth-capabilities messages specifying
+pre-shared key ease of input for the user and supported PSK input methods.
+The agent with the lowest PSK ease of input presents PSK to the user when the agent
```
Comment (pthatcherg) [inline index.bs]: "presents PSK" => "presents a PSK"

File: index.bs [hunk 5]
```diff
@@ -401,6 +401,25 @@ Issue(139): Clarify scoping/uniqueness of request IDs.
...
 
+Each supported authentication method is implemeted via authentication messages
+specific to that method.  Authentication method is explicitly specified by
+the message itself.  Authentication status message is common for all authentication
+methods.  Any new authentication method added must define new authentication messages.
+Default authentication method is a challenge-response authentication with
+auth-request-hkdf-scrypt-psk and auth-response-hkdf-scrypt-psk-result.
+
+Prior to authentication, agents exchange auth-capabilities messages specifying
+pre-shared key ease of input for the user and supported PSK input methods.
```
Comment (pthatcherg) [inline index.bs]: Maybe put "pre-shared key (PSK)" to make it more clear what "PSK" means.

File: index.bs [hunk 6]
```diff
@@ -401,6 +401,25 @@ Issue(139): Clarify scoping/uniqueness of request IDs.
...
+The agent with the lowest PSK ease of input presents PSK to the user when the agent
+either sends or receives an authentication request.  In case both agents have the same
+PSK ease of input value, the receiver presents PSK to the user.  The same pre-shared key
```
Comment (pthatcherg) [inline index.bs]: "presents PSK" => "presents the PSK"

File: index.bs [hunk 7]
```diff
@@ -401,6 +401,25 @@ Issue(139): Clarify scoping/uniqueness of request IDs.
...
+is used by both agents to issue an authentication request.
+
+PSK ease of input is an integer in [0, 100] range, where 0 means it is not possible for
```
Comment (pthatcherg) [inline index.bs]: "in the range [0, 100]"

And perhaps be explicit that about it being inclusive.

File: index.bs [hunk 8]
```diff
@@ -401,6 +401,25 @@ Issue(139): Clarify scoping/uniqueness of request IDs.
...
+the user to input PSK on this device and 100 means that it's easy for the user to
+input PSK on the device.  Supported PSK input methods are numeric, alphanumeric and
```
Comment (pthatcherg) [inline index.bs]: Perhaps we should just go with numeric given the feedback we got about asian languages.

File: index.bs [hunk 9]
```diff
@@ -465,16 +484,15 @@ For hkdf-of-scrypt-of-psk, the proof is calculated using the following steps:
 
 8. Let salt be the salt from the authentication-request message.
 
-9. Let info be a CBOR-serialized certificate-fingerprint-pair object (CDDL
-     defined in Appendix A) with the following values:
+9. Let info be a 64 byte array containing certificate fingerprint pair with the following values:
 
--   challenger-fingerprint: The result of running sha-256 on the
-    Distinguished Encoding Rules (DER) form (see
+-   Bytes 0-31 of the array are challenger fingerprint: The result of running
```
Comment (pthatcherg) [inline index.bs]: "are the challenger's fingerprint"

File: index.bs [hunk 10]
```diff
@@ -465,16 +484,15 @@ For hkdf-of-scrypt-of-psk, the proof is calculated using the following steps:
...
+    sha-256 on the Distinguished Encoding Rules (DER) form (see
     https://tools.ietf.org/html/rfc8122#section-5) of the certificate used by
     the challenger in the QUIC crypto handshake during connection establishment.
 
--   responder-fingerprint: The result of running sha-256 on the
-    Distinguished Encoding Rules (DER) form (see
+-   Bytes 32-63 of the array are responder fingerprint: The result of running
```
Comment (pthatcherg) [inline index.bs]: "are the responder's fingerprint"

File: index.bs (hunk 8 above)
Comment (maxyakimakha) [reply index.bs]: Leaving it as is, up to the implementer if they choose to support and advertise it.

**Pull request merged**